Add Event Cards

purna135 opened the pull request.

canyon289 commented:
This is amazing, ill do a full review in next 8 hours. One question, can the colors for upcoming and past events be made the same?

<img width="1189" alt="image" src="https://user-images.githubusercontent.com/7213793/211204001-148d024d-b5c2-44c9-8b5c-03a5dfbdddee.png">
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/7213793/211204041-6a2a33c5-7d75-46c1-801a-b58de7569a0b.png">

purna135 commented:
Sure, which colour do you prefer? 😀

canyon289 commented:
Upcoming seems nice

purna135 commented:
Done, additionally, I updated the button's hover colour from red to the same bluish color; let me know which one you think looks best.

canyon289 commented on layouts/index.html:
```diff
@@ -59,38 +59,64 @@ <h3> {{ .title }} </h3>
-                        </article>
-                    {{ end }}
+    <!-- ================================== START OF FEATURED EVENT ============================== -->
+    {{ $events := sort .Site.Data.events.event "eventOrder" }}
```
<img width="1462" alt="image" src="https://user-images.githubusercontent.com/7213793/211214892-726264de-7fcc-40eb-afce-2d680eb4434b.png">

Can events be moved above goals?

purna135 replied:
Here is a preview of both the style, could you please take a look and let me know one looks better?

Event at the top of the goals: https://purna135.github.io/pymcon-test/
Event at the bottom of the goals: https://deploy-preview-60--pymcon.netlify.app/

canyon289 replied:
Top of goals is better

canyon289 commented on data/events.toml:
```diff
@@ -0,0 +1,63 @@
+    dataTime = "Thu Nov 17, 2022, 8:30 PM IST"
+    title = "Bayesian Modeling in Biotech: Using PyMC to Analyze Agricultural Data."
+    place = "Online"
+    description = "In this panel discussion we will discuss why Bayesian modeling is such a powerful tool for solving problems in biotechnology. As experiments are often complex it is important to build custom and causal models that accurately represent the structure of the experiment in the statistical model. In this panel discussion we will discuss why Bayesian modeling is such a powerful tool  ...
```
Is there a way to have the box always expanded rather than expand over scrollover

<img width="1328" alt="image" src="https://user-images.githubusercontent.com/7213793/211214969-52072799-89f8-44a0-83ce-16b0f24d7b9b.png">

purna135 replied:
I had the same thought at first. I then consider that the description may be too long, affecting the consistent size of the cards, so I create them to show few lines and expand on user interaction.
but I have now corrected them.

However, this effect remains for event headings of featured events; it will only show the first two lines and expand the rest based on user interaction(you can check one of the featured event card); let me know if you want to expand that as well.

canyon289 replied:
got it, I trust you thought about it more than me. Thank you for explaining it. No need for a change

canyon289 replied:
And thanks for considering all the options

purna135 commented on data/events.toml:
```diff
@@ -0,0 +1,63 @@
+# eventOrder : integer (used to order events ascending order)
```
Here is a preview of both the style, could you please take a look and let me know one looks better?

Event at the top of the goals: https://purna135.github.io/pymcon-test/
Event at the bottom of the goals: https://deploy-preview-60--pymcon.netlify.app/

canyon289 commented:
This is great! My vote is to merge this but after disabling autodeploy in CI. That way we can then update the website with the latest event details and such

OriolAbril commented:
> This is great! My vote is to merge this but after disabling autodeploy in CI. That way we can then update the website with the latest event details and such

You could change the `on push main` trigger for `workflow_dispatch` to be able to trigger it manually but still have everything be done with github actions and still build but not deploy in previews to catch potential errors. Example: https://github.com/pymc-devs/pymc-data-umbrella/blob/main/.github/workflows/tx-pull.yml#L6

purna135 commented:
> You could change the [`on push main`](https://github.com/pymc-devs/pymcon_web_series_website/blob/main/.github/workflows/gh-pages.yml#L4-L6) trigger for `workflow_dispatch` to be able to trigger it manually but still have everything be done with github actions and still build but not deploy in previews to catch potential errors. Example: https://github.com/pymc-devs/pymc-data-umbrella/blob/main/.github/workflows/tx-pull.yml#L6

This is really great.


> merge this but after disabling autodeploy in CI.

I don't think we need to disable it; I'll remove the demo content in my next commit and add a few conditions for no events.

purna135 commented:
> Regarding the event page, I feel past events are a bit too hidden. Would it be possible to keep the right side sidebar but have the elements act more as a table of contents thing? e.g. all events are there, sorted chronologically with past ones at the bottom, and there is a "past events" heading separating upcoming from past events which is where the past events text on the right sidebar links to.


Could you please check now that I have added a new button for all of the events?

Let me know if you want to fix the position of the right side buttons on scroll.

purna135 commented:
> * My main issue is with the "scroll/move left animation". I like to take my time doing things but I am not super slow either. I am unable to read the title plus description of the latest event before it disappears from view, to me this removes attention from the event appearing first when I think we'd want that to be the most eye catching.

The "scroll/move left animation" will be enabled only when there are more than three featured events; will there be more than three featured events in the future? If so, I will fix this issue.

canyon289 commented:
Thanks @OriolAbril and @purna135. I trust between you both these details can get figured out. I left some comments below

> My main issue is with the "scroll/move left animation". I like to take my time doing things but I am not super slow either. I am unable to read the title plus description of the latest event before it disappears from view, to me this removes attention from the event appearing first when I think we'd want that to be the most eye catching.

We shouldn't have more than 3 at a time. Either way its a setting in the JS library to disable it

> I don't understand the order event cards appear in the homepage. I think that should be chronological but it doesn't look like it.
I believe this this is coming from the event order setting in the toml which is what I asked. I think this is demo content right now If so we can modify it right @purna135?

> Whole card click on front page
This would be great actually. If possible can we do this?


> I don't think we need to disable it; I'll remove the demo content in my next commit and add a few conditions for no events.

You've already done an incredible job on this website building the framework. If you'd like I'd be happy to ask the event mentors to add their events themselves. Let me know

purna135 commented:
> I don't understand the order event cards appear in the homepage. I think that should be chronological but it doesn't look like it.


> I believe this this is coming from the event order setting in the toml which is what I asked. I think this is demo content right now If so we can modify it right @purna135?


Yes, we are sorting the events in ascending order based on the "eventOrder" property.

purna135 commented:
> I still see the goals before the events though, we should update before merging so we don't forget.

✅ Done



> * My main issue is with the "scroll/move left animation".

✅ Done, because we only display three events as featured events, the scroll animation will not appear, and even if we increase the number of featured events in the future, I have fixed the scroll animations.



> * Even if I hover and activate the animation for the card background, the scrolling out of view does not stop.

✅ Fixed, The animation will now be paused on mouse hover, mouse click, and keyboard focus.




> * I don't understand the order event cards appear in the homepage. I think that should be chronological but it doesn't look like it.

✅ We have a "eventOrder" property that allows us to control the order in which events are displayed.



> * I want to click on the card so badly! But the link is only the learn more button.

✅ We can now click on the entire card, which will take us to an external event post.



> * render the author as a link to their website of their preferred social link.

✅ Done, now we can click on the Author image/name.



> I feel past events are a bit too hidden

✅ We now have three buttons: all events, upcoming events, and past events (all events is the default), also made the buttons sticky on scroll.




👉**NOTE:**
I've added some conditional statements that keep the current layout unchanged when there are no events to display, so we can merge this PR without breaking the autodeploy CI.

👉**LAST STEP:**
Once we've verified that everything is in order, we can remove the dummy data from event.toml and merge this PR to begin filling out the event details in separate PRs.

Also, I used some demo headings and descriptions (for example, PyMCon Events and its description on the event page), so please suggest some better headlines.


@canyon289 @mjhajharia @OriolAbril Please take a look and make any suggestions for improvement.

Thank you 😊

canyon289 commented:
No action needed on this comment. i checked the website for accessibility. Were good here. Thank you Purna for handling this

<img width="1441" alt="image" src="https://user-images.githubusercontent.com/7213793/211677988-84899168-b0f6-474f-843d-a04eba10aa0b.png">

purna135 commented:
Agreed, I removed the read more button from the hover effect, but I think it's better to remove the hover effect altogether, what do you think? Could you please take a look now?

purna135 commented:
Agreed, I removed the read more button from the hover effect, but I think it's better to remove the hover effect altogether, what do you think? Could you please take a look now?

purna135 commented:
I removed the hover effect because there is a read more button to get more details.

cluhmann commented:
Yeah, that works.  I think the cards look great.  In terms of keeping a uniform size across events, I think we can edit the short bit of text for future events so that it matches the existing events and helps to maintain equivalent sizing.

purna135 commented:
> I think we can edit the short bit of text for future events so that it matches the existing events

There won't be a problem with the card's size because the extra information will be hidden and just the text that fits the space will be displayed.

cluhmann commented:
Even better!

purna135 commented:
Is there anything else we need to do with this PR?

canyon289 commented:
For now I dont think so. Thank you I just need to get event cards up. i can do so this week

purna135 commented:
Thank you for your confirmation; let me to remove the dummy event data so that we can merge this PR.

canyon289 commented:
I created another PR against your branch to include an actual event card as well

canyon289 commented:
@purna135 thanks much for this PR I'll detail specifically what I think the next sequence should be

1. Merge this PR from me into this branch so we can inspect the final result https://github.com/purna135/pymcon_2022_website/pull/1
  * Also then once we merge we'll have an actual event on the page
2. Each event mentor can follow up with their own PR to add their event

WDYT?

canyon289 commented on data/events.toml:
```diff
@@ -0,0 +1,33 @@
+    readMoreURL = "https://www.meetup.com/pymc-online-meetup/events/287214141/"
+    eventOrder = 1
+    eventType = "upcoming"
+    featured = true
```
@purna135  Do you know why the event is not showing up on the home page even though its featured/

canyon289 commented:
Thanks Purna
One last question, do you know where we can replace the default speaker image with one thats more gender and skin color neutral? Dont feel obligated to do this yourself, youve already done great work on this PR. just want your guidance on how we can go about this for DEi



<img width="343" alt="image" src="https://user-images.githubusercontent.com/7213793/212789994-5a933d34-c7bf-40f9-8324-6e149e13f860.png">

purna135 commented on layouts/index.html:
```diff
@@ -35,8 +35,83 @@ <h1>{{.Params.Title}}</h1>
+            {{ end }}
+        {{ end }}
+
+        <!-- Only display the "Featured event section" if there are more than two events to display. -->
+        {{ if gt $count 2 }}
+            {{ $showFeaturedEvent = true }}
+        {{ end }}
+    {{ end }}
```
Currently, we display the featured event if there are at least three events to display; please let me know if this needs to be changed, and I will make the necessary changes.

canyon289 replied:
ah, Yes if possible please make it show up even if theres just one

purna135 replied:
ok

purna135 commented on data/events.toml:
```diff
@@ -0,0 +1,33 @@
+#    readMoreURL = "https://www.meetup.com"          (Event post URL i.e Discourse, Meetup etc)
+#    eventOrder = 2                                  (An integer, used to order events in ascending order)
+#    eventType = "upcoming"                          (must be "upcoming" or "past")
+#    featured = true                                 (A boolean, if true, the event will be displayed on home pade NOTE: We only display the "Featured event section" if there are more than three events to display.)
```
Here is a note for the same

canyon289 replied:
ah im sorry I should have read the documentation

purna135 commented:
> do you know where we can replace the default speaker image with one thats more gender and skin color neutral?

[pymcon_web_series_website](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3)/[static](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3/static)/[images](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3/static/images)/[speakers](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3/static/images/speakers)

here we have a default image.
I recently updated the default image, could you please verify if it is suitable for use?

canyon289 commented:
> > do you know where we can replace the default speaker image with one thats more gender and skin color neutral?
> 
> [pymcon_web_series_website](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3)/[static](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3/static)/[images](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3/static/images)/[speakers](https://github.com/pymc-devs/pymcon_web_series_website/tree/31bb8e3743876e6acd80ab2dbc0e45b29cde14b3/static/images/speakers)
> 
> here we have a default image. I recently updated the default image, could you please verify if it is suitable for use?

Yes the image is perfect,

For the featured image, allowing for them to show up if there's at least 1 would be fantastic. Most of the time we'll only have 1 to 3 as is.

cluhmann commented:
This is looking good to me.  I suspect that we'll be able to figure out things to tweak once we start adding in real events.  Do we want the first event to be cleaned up before merging?

cluhmann commented:
Yup, I think that makes sense.

cluhmann commented:
How is this PR looking to you @canyon289 ? Anything left to do before merging or can we merge it and then start adding the real info for specific events?

canyon289 commented:
We want featured events to show up even if theres less than three but we can do that on a follow up. Merging this as is

canyon289 commented:
Thank you very much @purna135 for getting this together for us
